src/guards/book-exists.ts

```diff
@@ -50,7 +49,7 @@ export class BookExistsGuard implements CanActivate {
    */
   hasBookInApi(id: string) {
     return this.googleBooks.retrieveBook(id)
-      .map(book => this.bookActions.loadBook(book))
+      .map(book => new bookActions.LoadBook(book))
       .do(action => this.store.dispatch(action))
       .map(book => !!book)
       .catch(() => {
```

Comment:
The angular/router differs from how ngrx/router works in this regard. Returning false cancels navigation altogether, where ngrx/router would just not consider the route path to be a candidate anymore. That's why navigation is required or it just wouldn't go anywhere.